pkg/reconciler/taskrun/validate_taskrun.go

```diff
@@ -159,8 +162,12 @@ func findMissingKeys(neededKeys, providedKeys map[string][]string) map[string][]
 // ValidateResolvedTask validates that all parameters declared in the TaskSpec are present in the taskrun
 // It also validates that all parameters have values, parameter types match the specified type and
 // object params have all the keys required
-func ValidateResolvedTask(ctx context.Context, params []v1beta1.Param, matrix *v1beta1.Matrix, rtr *resources.ResolvedTask) error {
-	if err := validateParams(ctx, rtr.TaskSpec.Params, params, matrix.GetAllParams()); err != nil {
+func ValidateResolvedTask(ctx context.Context, params v1beta1.Params, matrix *v1beta1.Matrix, rtr *resources.ResolvedTask) error {
```

Comment:
@dibyom Sorry, disregard. The existing logic is correct. I was confused why we were hardcoding the type string, but it's because it is checking the type for string replacements not the param type itself. Added a comment to clarify why we are checking type string.